### PR TITLE
Hf referral new digit component integration

### PR DESCRIPTION
**Pull request by Shashwat12-egov**

### CODE REVIEW

File: packages/referral_reconciliation/pubspec.lock
Comment (naveenr-egov) [inline packages/referral_reconciliation/pubspec.lock:1]: Why is this deleted?

**Review by naveenr-egov — CHANGES_REQUESTED** [review]
Resolve the comments. Use dart code format before commiting.

File: packages/referral_reconciliation/assets/animated_json/alert.json
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/assets/animated_json/alert.json:1]: On removing from any of them it is giving error

File: packages/referral_reconciliation/lib/pages/create_referral/record_facility_details.dart [hunk 1]
```diff
@@ -24,7 +25,7 @@ class ReferralFacilityPage extends LocalizedStatefulWidget {
   const ReferralFacilityPage({
     super.key,
     super.appLocalizations,
-    this.isEditing = false,
+    this.isEditing = false,show
```
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/lib/pages/create_referral/record_facility_details.dart:28]: I added to control the display property of the textfields but now it is fixed. I have removed it.

File: packages/referral_reconciliation/lib/pages/project_facility/project_facility_selection.dart [hunk 2]
```diff
@@ -134,14 +140,15 @@ class _ReferralReconProjectFacilitySelectionPageState
...
+                                margin: EdgeInsets.only(
+                                  top: theme.spacerTheme.spacer2,
+                                  left: theme.spacerTheme.spacer2,
+                                  right: theme.spacerTheme.spacer2,
                                 ),
                                 decoration: BoxDecoration(
-                                  color:
-                                      DigitTheme.instance.colors.alabasterWhite,
+                                  color: Colors.white,
+                                  // color:
```
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/lib/pages/project_facility/project_facility_selection.dart:150]: Done

File: apps/health_campaign_field_worker_app/pubspec.lock
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply apps/health_campaign_field_worker_app/pubspec.lock:1]: Added

File: packages/attendance_management/pubspec.lock
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/attendance_management/pubspec.lock:1]: Added

File: packages/referral_reconciliation/pubspec.lock
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/pubspec.lock:1]: added

File: packages/referral_reconciliation/lib/pages/create_referral/reason_checklist_preview.dart [hunk 3]
```diff
@@ -43,18 +44,21 @@ class _ReferralReasonChecklistPreviewPageState
...
+                          size: ButtonSize.large,
+                          label: localizations.translate(i18.common.corecommonclose),
+                          mainAxisSize: MainAxisSize.max,
                           onPressed: () {
                             context.read<ServiceBloc>().add(
-                                  ServiceResetEvent(serviceList: serviceList),
-                                );
+                              ServiceResetEvent(serviceList: serviceList),
+                            );
+                            context.router.popUntil((route) => route.settings.name == 'SearchReferralReconciliationsRoute');
```
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/lib/pages/create_referral/reason_checklist_preview.dart:58]: Changed the hard coded route name.
I have added the navigation here to navigate to the main page on clicking close button

File: packages/referral_reconciliation/lib/pages/create_referral/record_reason_checklist.dart [hunk 4]
```diff
@@ -539,98 +573,114 @@ class _ReferralReasonChecklistPageState
...
-          inputFormatter: [
-            FilteringTextInputFormatter.allow(RegExp(
-              "[a-zA-Z0-9 ]",
-            )),
-          ],
+        child: FormField<String>(
           validator: (value) {
-            if (((value == null || value == '') && item.required == true)) {
+            // Custom validation logic
+            if (((controller[index].text == '') && item.required == true)) {
```
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/lib/pages/create_referral/record_reason_checklist.dart:579]: Previously, we were using custom flutter component so it was populating the value with the textfield but now value is coming null so instead of value we are directly taking the value from form.

File: packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart [hunk 5]
```diff
@@ -405,65 +126,51 @@ class _RecordReferralDetailsPageState
...
-                                                  final symptom = form
-                                                      .control(_referralReason)
-                                                      .value as String;
-                                                  if (value1.isNotEmpty) {
+                                                  if (viewOnly) {
+                                                    final symptom = form
+                                                        .control(
+                                                            _referralReason)
+                                                        .value as String;
                                                     context
```
[the remark this answers is not in the document]
Comment (Shashwat12-egov) [reply packages/referral_reconciliation/lib/pages/create_referral/record_referral_details.dart:165]: Added

**Comment by Shashwat12-egov** [comment]
Raised PR from different branch ([hf-referral-new-digit-component](https://github.com/egovernments/health-campaign-field-worker-app/tree/hf-referral-new-digit-component)).

**Comment by Shashwat12-egov** [comment]
Raised PR with different branch [hf-referral-new-digit-component](https://github.com/egovernments/health-campaign-field-worker-app/tree/hf-referral-new-digit-component)
